ci/diff.py

```diff
@@ -1,25 +1,36 @@
+from __future__ import annotations
+
 import difflib
 import json
 import re
 import sys
 from typing import Any, Dict, List
 
+timestamp_pattern = re.compile(
+    r"\d{4}-\d{2}-\d{2}[ T]\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:[+-]\d{2}:\d{2})?|\d{4}-\d{2}-\d{2}"
+)
+datetime_str_pattern = re.compile(
+    r"\b[A-Za-z]{3}\s[A-Za-z]{3}\s([0-2]?[0-9]|3[0-1])\s([0-1]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])\s[0-9]{4}\b"
+)
+# datetime_str_pattern = re.compile(
+#     r"\b(Sun|Mon|Tue|Wed|Thu|Fri|Sat)\s(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s([0-2]?[0-9]|3[0-1])\s([0-1]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])\s[0-9]{4}\b"
+# )
```

Comment:
Can probably remove this if we decide to use the other (more lenient) approach

Reply:
Probably, but I'd need to read regular expressions again, so I'm inclined to leave it for now...